### PR TITLE
Add search field style.

### PR DESCRIPTION
**Pull request by pixlwave**
Also adds `Compound.configureSystemComponents()` to apply some tweaks that can only be done with global appearances.

### CODE REVIEW

File: Sources/Compound/Compound.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+public struct Compound {
+    /// Themes some default components to match Compound as needed.
+    public static func theme() {
```
Comment (pixlwave) [inline Sources/Compound/Compound.swift:21]: I don't think this is the right name, but struggling to think of a better one right now 😕

Comment (stefanceriu) [reply Sources/Compound/Compound.swift:21]: How about `configureTheme`?

File: Sources/Compound/Text Field Styles/SearchableStyle.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            clearButton?.setImage(buttonImage, for: .normal)
+            clearButton?.tintColor = placeholderColor
+            
+            // Text field.
+            searchTextField.textColor = textColor
+            searchTextField.backgroundColor = textFieldBackgroundColor
+            searchTextField.tintColor = textFieldTintColor
+            
+            // Hide the effect views so we can the rounded rect style without any materials.
+            let effectBackgroundTop = searchTextField.value(forKey: "_effectBackgroundTop") as? UIView
```
Comment (stefanceriu) [inline Sources/Compound/Text Field Styles/SearchableStyle.swift:53]: 🙈

**Review by stefanceriu — APPROVED** [review]
lgtm 👍

**Pull request merged**